articles/tenntenn.re

```diff
@@ -0,0 +1,668 @@
+  :-1        0x104ea98  cc                  INT $0x3
+  :-1        0x104ea99  cc                  INT $0x3
+  :-1        0x104ea9a  cc                  INT $0x3
+  :-1        0x104ea9b  cc                  INT $0x3
+  :-1        0x104ea9c  cc                  INT $0x3
+  :-1        0x104ea9d  cc                  INT $0x3
+  :-1        0x104ea9e  cc                  INT $0x3
+  :-1        0x104ea9f  cc                  INT $0x3
+//}
+
+なお、@<code>{go1.12.17}コマンドは@<code>{go}コマンドのバージョンが1.12.17であることを示しています。
+バージョンごとのGoのツールチェインをインストールしたい場合、
+は@<code>{go get golang.org/dl/go1.12.17}を実行し、@<code>{go1.12.17 download}と実行することで
+必要なものがダウンロードとインストールされます。
+
+Go1.12では、@<code>{runtime.deferprocStack}関数が導入されていないため、@<code>{runtime.deferproc}関数が呼ばれていることが分かります。
+
+一方、Go1.13のコンパイラでコンパイルすると、@<code>{runtime.deferprocStack}関数が用いられていることが分かります。
+
+//cmd{
+$ go1.13.8 build -o main.o /tmp/main.go
+$ go1.13.8 tool objdump -s "main\.main$" main.o
+TEXT main.main(SB) /tmp/main.go
+  main.go:3  0x1051560  65488b0c2530000000  MOVQ GS:0x30, CX
+  main.go:3  0x1051569  483b6110            CMPQ 0x10(CX), SP
+  main.go:3  0x105156d  7669                JBE 0x10515d8
+  main.go:3  0x105156f  4883ec58            SUBQ $0x58, SP
+  main.go:3  0x1051573  48896c2450          MOVQ BP, 0x50(SP)
+  main.go:3  0x1051578  488d6c2450          LEAQ 0x50(SP), BP
+  main.go:4  0x105157d  c744241010000000    MOVL $0x10, 0x10(SP)
+  main.go:4  0x1051585  488d057c440200      LEAQ go.func.*+64(SB), AX
+  main.go:4  0x105158c  4889442428          MOVQ AX, 0x28(SP)
+  main.go:4  0x1051591  488d0575fd0100      LEAQ go.string.*+237(SB), AX
+  main.go:4  0x1051598  4889442440          MOVQ AX, 0x40(SP)
+  main.go:4  0x105159d  48c744244804000000  MOVQ $0x4, 0x48(SP)
+  main.go:4  0x10515a6  488d442410          LEAQ 0x10(SP), AX
+  main.go:4  0x10515ab  48890424            MOVQ AX, 0(SP)
+  main.go:4  0x10515af  e8dc1ffdff          CALL runtime.deferprocStack(SB)
+  main.go:4  0x10515b4  85c0                TESTL AX, AX
+  main.go:4  0x10515b6  7510                JNE 0x10515c8
+  main.go:5  0x10515b8  90                  NOPL
+  main.go:5  0x10515b9  e8d225fdff          CALL runtime.deferreturn(SB)
+  main.go:5  0x10515be  488b6c2450          MOVQ 0x50(SP), BP
+  main.go:5  0x10515c3  4883c458            ADDQ $0x58, SP
+  main.go:5  0x10515c7  c3                  RET
+  main.go:4  0x10515c8  90                  NOPL
+  main.go:4  0x10515c9  e8c225fdff          CALL runtime.deferreturn(SB)
+  main.go:4  0x10515ce  488b6c2450          MOVQ 0x50(SP), BP
+  main.go:4  0x10515d3  4883c458            ADDQ $0x58, SP
+  main.go:4  0x10515d7  c3                  RET
+  main.go:3  0x10515d8  e81381ffff          CALL runtime.morestack_noctxt(SB)
+  main.go:3  0x10515dd  eb81                JMP main.main(SB)
+  :-1        0x10515df  cc                  INT $0x3
+//}
+
+また、@<code>{runtime.deferproc}関数や@<code>{runtime.deferprocStack}関数はランタイムで提供されている関数であるため、次のようにコンパイラによって生成されるバイナリにも含まれています。
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Line 285 sentence length(102) exceeds the maximum sentence length of 100.
Over 2 characters. (ja-technical-writing/sentence-length)